File: 5.implementation/VT/redsun/src/main/webapp/static/js/services/WorkflowsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('workflowService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(workflow) {
			var urlServer = $rootScope.makePostURL("/workflow/insert");
			return $http.post(urlServer, workflow);
		}
		
		// Update.
		, update: function(workflowId, workflow) {
			var urlServer = $rootScope.makePostURL("/workflow/update/" + workflowId);
			return $http.put(urlServer, workflow);
		}
		
		// Delete.
		, delete: function(workflowId) {
			var urlServer = $rootScope.makePostURL("/workflow/delete/" + workflowId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(workflowId) {
			var url = $rootScope.makeGetURL("/workflow/getbyid/" + workflowId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, workflow) {
			var url = $rootScope.makeGetURL("/workflow/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: workflow });
		}
		
	}// return.
});

File: 5.implementation/VT/redsun-bpmn/src/main/java/com/redsun/bpmn/service/WorkflowsService.java
package com.redsun.bpmn.service;

import com.redsun.bpmn.entities.Workflows;
import com.redsun.bpmn.entities.Result;

/**
 * Workflows Service interface
 */
public interface WorkflowsService {
	
    // Insert.
	Result insert(final Workflows workflows) ;

    // Update.
	Result update(final Workflows workflows) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listWorkflowsForPageAndFilter(final int itemsPerPage, final int pageNo, final Workflows workflows);

}
